src/ui/molecules/header/Header.tsx

```diff
@@ -32,18 +51,110 @@ const BurgerMenu = ({
   </div>
 )
 
+const NavItem = ({
+  isItemSelected,
+  isRow,
+  item
+}: DeepReadonly<{
+  isItemSelected: boolean
+  isRow: boolean
+  item: JSX.Element
+}>): JSX.Element => (
+  <>
+    {isItemSelected && <Icon name={isRow ? 'arrow-down' : 'arrow-right'} size={15} />}
+    {item}
+    {isItemSelected && isRow && <Icon name="arrow-up" size={15} />}
+  </>
+)
+
+const SubMenu = ({
+  onSubItemSelection,
+  subMenu
+}: DeepReadonly<{
+  onSubItemSelection: () => void
+  subMenu: JSX.Element[]
+}>): JSX.Element => {
+  return (
+    <div className="okp4-header-navigation-submenu">
+      {subMenu.map((navItem: DeepReadonly<JSX.Element>, index: number) => (
+        <div
+          className="okp4-header-navigation-submenu-item"
+          key={index}
```

Comment:
Here, the use of the index as a key can cause issues [more info here](https://reactjs.org/docs/lists-and-keys.html).

Reply:
Yes I know, but I really don't see how to handle keys based on jsx elements that don't have ids or other values that I can use....

Reply:
Maybe you can use the `short.generate()` function as it's already imported in the file

Reply:
i thought about it, but I think it's useless... check the answer of this : https://stackoverflow.com/questions/52196127/different-ways-to-add-a-key-to-jsx-element-in-loop-in-react